src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -143,7 +143,7 @@ public string GenerateStartTag()
         /// </returns>
         public string GeneratePreContent()
         {
-            if (SelfClosing)
+            if (!string.IsNullOrWhiteSpace(TagName) && SelfClosing)
```

Comment:
reminded of requirement here offline.  change is :ok: but I still recommend the `<remarks/>` addition.